pkg/sdk/tasks_impl_gen.go

```diff
@@ -0,0 +1,219 @@
+package sdk
+
+import (
+	"context"
+)
+
+var _ Tasks = (*tasks)(nil)
+
+type tasks struct {
+	client *Client
+}
+
+func (v *tasks) Create(ctx context.Context, request *CreateTaskRequest) error {
+	opts := request.toOpts()
+	return validateAndExec(v.client, ctx, opts)
+}
+
+func (v *tasks) Alter(ctx context.Context, request *AlterTaskRequest) error {
+	opts := request.toOpts()
+	return validateAndExec(v.client, ctx, opts)
+}
+
+func (v *tasks) Drop(ctx context.Context, request *DropTaskRequest) error {
+	opts := request.toOpts()
+	return validateAndExec(v.client, ctx, opts)
+}
+
+func (v *tasks) Show(ctx context.Context, request *ShowTaskRequest) ([]Task, error) {
+	opts := request.toOpts()
+	dbRows, err := validateAndQuery[taskDBRow](v.client, ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+	resultList := convertRows[taskDBRow, Task](dbRows)
+	return resultList, nil
+}
+
+func (v *tasks) ShowByID(ctx context.Context, id SchemaObjectIdentifier) (*Task, error) {
+	return v.Describe(ctx, id)
+}
+
+func (v *tasks) Describe(ctx context.Context, id SchemaObjectIdentifier) (*Task, error) {
+	opts := &DescribeTaskOptions{
+		name: id,
+	}
+	result, err := validateAndQueryOne[taskDBRow](v.client, ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+	return result.convert(), nil
+}
+
+func (v *tasks) Execute(ctx context.Context, request *ExecuteTaskRequest) error {
+	opts := request.toOpts()
+	return validateAndExec(v.client, ctx, opts)
+}
+
+func (r *CreateTaskRequest) toOpts() *CreateTaskOptions {
```

Comment:
did you want to rename this to "CreateTaskOperation"? instead of Options

Reply:
maybe we can do this later

Reply:
I will add it to our bag of enhancements of SDK.

Reply:
Added as SNOW-933471.